### PR TITLE
nim: submission

### PR DESCRIPTION
**Pull request by gpanders**
#### Description

Add [nim](https://nim-lang.org) port.

Nim's install script wants to install everything under a `nim` subdirectory under whatever directory you provide to `install.sh`. So for example if we use `./install.sh /opt/local` this would create an `/opt/local/nim` directory, which we obviously don't want. So I copied what Go does and installed this directory under `/opt/local/lib` so that it creates `/opt/local/lib/nim`. Binaries (`nim`, `nimble`, `nimgrep`, and `nimsuggest`) are symlinked from that directory (e.g. `/opt/local/bin/nim -> /opt/local/lib/nim/bin/nim`).

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.6 18G8022
Xcode 11.3.1 11C505

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [X] tried existing tests with `sudo port test`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

### CODE REVIEW

File: lang/nim/Portfile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+PortSystem          1.0
+
+name                nim
+version             1.4.4
+revision            0
+license             MIT
+categories          lang
+platforms           darwin
+supported_archs     i386 x86_64 ppc64
```
Comment (herbygillot) [inline lang/nim/Portfile:11]: You would only set `supported_archs` if you know for sure that a port only works with a given subset of architectures.

In this case, it looks like `nim` supports the whole hog, including **arm64**. So remove this and the right thing should get passed automatically as per the local architecture.

File: lang/nim/Portfile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+checksums           rmd160  df2568109792112495aaed4273f222bf8623f95b \
+                    sha256  9f43f5da8f7130b137d59e49f3911f0c35cc1e94e68c333c50c59ef8b65dd6f6 \
+                    size    159095112
+
+use_configure       no
+
+variant doc description {Build HTML docs} {}
+
+build {
+    system -W ${worksrcpath} "./build.sh --os ${os.platform} --cpu ${os.arch}"
```
Comment (herbygillot) [inline lang/nim/Portfile:35]: Per my current understanding, you may want to use `${configure.build_arch}` here.

See https://github.com/macports/macports-ports/commit/84dd03127da4647acd1fea8d612dedde07c7e3f3

File: lang/nim/Portfile [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+description         Nim programming language
+long_description    Nim is a statically typed compiled systems programming \
+                    language. It combines successful concepts from mature \
+                    languages like Python, Ada, and Modula. Its design focuses \
+                    on efficiency, expressiveness, and elegance (in that \
+                    order of priority).
+
+homepage            https://nim-lang.org
+
+master_sites        https://nim-lang.org/download/
```
Comment (herbygillot) [inline lang/nim/Portfile:24]:
```
master_sites ${homepage}/download
```

**Review by herbygillot — CHANGES_REQUESTED** [review]

File: lang/nim/Portfile (hunk 2 above)
Comment (herbygillot) [reply lang/nim/Portfile:35]: Actually I may be wrong on this one, disregard.

File: lang/nim/Portfile (hunk 1 above)
Comment (gpanders) [reply lang/nim/Portfile:11]: I was going off of their [README](https://github.com/nim-lang/Nim#compiling) which lists only x86, x86_64, and ppc64 for macOS. That’s for stable/tested versions only, so I suppose it does technically build for arm64 but it’s not officially supported by the Nim project (yet). What’s the call here in this case? Do we list *all* archs even if not officially supported, or do we stick with what upstream says?

File: lang/nim/Portfile (hunk 3 above)
Comment (gpanders) [reply lang/nim/Portfile]: Fixed

File: lang/nim/Portfile (hunk 1 above)
Comment (herbygillot) [reply lang/nim/Portfile:11]: Ah, in that case we should stick with upstream, but still remember to open things up (remove `supported_archs`) if the next version has official support for ARM64.

**Review by herbygillot — APPROVED** [review]

**Pull request merged**